numba/core/typing/new_builtins.py

```diff
@@ -127,53 +131,506 @@ def generic(self, args, kws):
+    # If everything is a NumPy integer, promotion should be to the
     # respective NumPy type.
-    if any('np' in tp.name for tp in inputs):
+    if all('np' in tp.name for tp in (arg_1, other)):
+        bitwidth = max(arg_1.bitwidth, other.bitwidth)
+        signed = arg_1.signed or other.signed
         return types.NumPyInteger.from_bitwidth(bitwidth, signed)
-
+    
     return types.py_int
 
-all_ints = (
-    sorted(set((types.py_int, types.py_int))) +
-    sorted(set((types.np_int32, types.np_int64))) +
-    sorted(set((types.np_uint32, types.np_uint64)))
-    )
-integer_binop_cases = tuple(
-    signature(choose_result_int(op1, op2), op1, op2)
-    for op1, op2 in itertools.product(all_ints, all_ints)
-    )
 
 class BinOp(ConcreteTemplate):
-    cases = list(integer_binop_cases)
-
+    cases = []
+ 
+    # Python cases
     cases += [signature(op, op, op) for op in sorted(types.py_real_domain)]
     cases += [signature(op, op, op) for op in sorted(types.py_complex_domain)]
+
+    # NumPy cases
     cases += [signature(op, op, op) for op in sorted(types.np_real_domain)]
     cases += [signature(op, op, op) for op in sorted(types.np_complex_domain)]
+    # TODO: Think about mixture float and complex cases
+    pass
 
 
-@infer_global(operator.add)
-class BinOpAdd(BinOp):
-    pass
+@overload(operator.is_)
+def ol_is_NotImplemented(a, b, /):
+    a_is_notimplemented_ty = isinstance(a, types.NotImplementedType)
+    b_is_notimplemented_ty = isinstance(b, types.NotImplementedType)
 
+    # only use this overload if at least one of the types is NotImplementedType
+    if a_is_notimplemented_ty or b_is_notimplemented_ty:
+        pred = a_is_notimplemented_ty and b_is_notimplemented_ty
+        def ol_is_NotImplemented(a, b, /):
+            return pred
+        return ol_is_NotImplemented
 
-@infer_global(operator.iadd)
-class BinOpAdd(BinOp):
-    pass
+
+@overload(operator.is_not)
+def ol_is_not_NotImplemented(a, b, /):
+    a_is_notimplemented_ty = isinstance(a, types.NotImplementedType)
+    b_is_notimplemented_ty = isinstance(b, types.NotImplementedType)
+
+    # only use this overload if at least one of the types is NotImplementedType
+    if a_is_notimplemented_ty or b_is_notimplemented_ty:
+        pred = not(a_is_notimplemented_ty and b_is_notimplemented_ty)
+        def ol_is_not_NotImplemented(a, b, /):
+            return pred
+        return ol_is_not_NotImplemented
+
+
+def bool_add_bool(x, y):
+    return NotImplemented
+
+
+def int_add_int(x, y):
+    return NotImplemented
+
+
+def float_add_float(x, y):
+    return NotImplemented
+
+
+def complex_add_complex(x, y):
+    return NotImplemented
+
+
+@intrinsic
+def intrin_bool_add_bool(tyctx, boolxty, boolyty):
+    assert boolxty == boolyty
+    sig = types.py_int(boolxty, boolyty)
+    def codegen(cgctx, builder, sig, llargs):
+        new_args = [cgctx.cast(builder, v, t, sig.return_type) for v, t in zip(llargs, sig.args)]
```

Comment:
PEP8 error. line too long.

I suggest enabling flake8 on this file since a lot of the content will be new.

Reply:
should avoid adding new flake8 errors